tests/automatic_calibration/test_calibration_node.py

```diff
@@ -0,0 +1,867 @@
+"""Test for the `CalibrationNode` class"""
+import os
+from datetime import datetime
+from io import StringIO
+from typing import Callable
+from unittest.mock import patch
+
+import numpy as np
+import pytest
+
+from qililab.automatic_calibration.calibration_node import (
+    CalibrationNode,
+    IncorrectCalibrationOutput,
+    export_calibration_outputs,
+)
+
+#################################################################################
+#################################### SET UPS ####################################
+#################################################################################
+
+logger_output_start = "RAND_INT:47102512880765720413 - OUTPUTS: "
+
+
+def dummy_comparison_model():
+    """Dummy comparison model, to provide to the fixtures."""
+    pass
+
+
+####################
+### MOCKED NODES ###
+####################
+@pytest.fixture(name="initialize_node_no_optional")
+@patch(
+    "qililab.automatic_calibration.calibration_node.CalibrationNode._build_notebooks_logger_stream",
+    return_value=StringIO(),
+)
+def fixture_initialize_node_no_optional(mocked_build_stream) -> CalibrationNode:
+    """Return a mocked CalibrationNode object for initialization, with the minimum number of things specified or mocked."""
+    return CalibrationNode(
+        nb_path="tests/automatic_calibration/notebook_test/zeroth.ipynb",
+        in_spec_threshold=0.6,
+        bad_data_threshold=0.9,
+        comparison_model=dummy_comparison_model,
+        drift_timeout=100.0,
+    )
+
+
+@pytest.fixture(name="initialize_node_optional")
+@patch(
+    "qililab.automatic_calibration.calibration_node.CalibrationNode._path_to_name_and_folder",
+    return_value=("node_id", "nb_folder"),
+)
+@patch(
+    "qililab.automatic_calibration.calibration_node.CalibrationNode.get_last_calibrated_output_parameters",
+    return_value={},
+)
+@patch(
+    "qililab.automatic_calibration.calibration_node.CalibrationNode.get_last_calibrated_timestamp",
+    return_value=0.0,
+)
+@patch(
+    "qililab.automatic_calibration.calibration_node.CalibrationNode._build_notebooks_logger_stream",
+    return_value=StringIO(),
+)
+def fixture_initialize_node_optional(
+    mocked_build_stream, mocked_last_cal_time, mocked_last_cal_params, mocked_path_to_folder
```

Comment:
unused mocked_build_stream, avoid warning by including the pylint disabled statement

Reply:
Done!